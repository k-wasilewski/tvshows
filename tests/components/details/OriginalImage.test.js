import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import ConnectedOriginalImage, {OriginalImage} from "../../../src/components/details/OriginalImage";
import {Provider} from "react-redux";
import {configure, mount} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";

describe("OriginalImage rendering specification", () => {
    it('OriginalImage is rendered', () => {
        const mockSrc = 'mockSrc';
        const mockTitle = 'mockTitle';

        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <ConnectedOriginalImage src={mockSrc} title={mockTitle} />
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("OriginalImage functional specification", () => {

    const getMaterialUIClassName = (className) => {
        switch (className) {
            case 'imageCard':
                return `makeStyles-${className}-17`;
            case 'imageWrapperCard':
                return `makeStyles-${className}-16`;
            case 'hidden':
                return `makeStyles-${className}-7`;
        }
    }

    it('renders img with src and title passed as props', () => {
        configure({adapter: new Adapter()});

        const mockSrc = 'mockSrc';
        const mockTitle = 'mockTitle';

        const component = mount(
            <OriginalImage src={mockSrc} title={mockTitle} />
        );

        const img = component.find(`.${getMaterialUIClassName('imageCard')}`).at(0);

        expect(img.prop('image')).toBe(mockSrc);
        expect(img.prop('title')).toBe(`${mockTitle}-img`);
        component.unmount();
    });

    it('is hidden when props.src value is empty', (done) => {
        configure({adapter: new Adapter()});

        const mockSrc = 'mockSrc';
        const mockTitle = 'mockTitle';

        const component = mount(
            <OriginalImage src={mockSrc} title={mockTitle} />
        );

        let imgCard = component.find('#originalImageCard').at(0);

        expect(imgCard.prop('className')).toBe(getMaterialUIClassName('imageWrapperCard'));

        component.setProps({src: ''});
        component.update();

        setTimeout(function () {
            imgCard = component.find('#originalImageCard').at(0);
            expect(imgCard.prop('className')).toBe(getMaterialUIClassName('hidden'));
            component.unmount();
            done();
        }, 500);
    });
});