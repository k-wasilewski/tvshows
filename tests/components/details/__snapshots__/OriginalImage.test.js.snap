// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OriginalImage rendering specification OriginalImage is rendered 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-hidden-7 MuiPaper-elevation1 MuiPaper-rounded"
  id="originalImageCard"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        
      </h2>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Zamknij
      </span>
    </button>
  </div>
</div>
`;
